syntax = "proto3";

package syft.core.store;

import "proto/core/common/common_object.proto";
import "proto/core/store/store_object.proto";
import "google/protobuf/any.proto";

message Dataset {
  syft.core.common.UID id = 1;
  string obj_type = 2;
  string schematic_qualname = 3;
  repeated syft.core.store.StorableObject data = 4;
  string description = 5;
  repeated string tags = 6;
  // fix by making Dict not StorableObject
  bytes read_permissions = 7;
  // fix by making Dict not StorableObject
  bytes search_permissions = 8;
}
